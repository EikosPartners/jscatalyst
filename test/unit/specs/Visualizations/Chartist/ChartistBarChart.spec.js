import Component from '@/components/Visualizations/Chartist/ChartistBarChart'
import Vue from 'vue'
import { shallow, mount } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'
import ChartistHelper from '@/common/chartistHelper.js'

const data = [ {"x": "5/15 - 6/14", "y": 13.3},
{"x": "6/14 - 7/14", "y": 12.08},
{"x": "7/14 - 8/13", "y": 14.62},
{"x": "8/13 - 9/12", "y": 17.57},
{"x": "9/12 - 10/12", "y": 14.35}  ]

describe('Chartist Bar', () => {
    let wrapper
    beforeEach(() => {
      wrapper = mount(Component, {
          propsData: {
              dataModel: data
         },
         attachToDocument: true
      })
    })

    afterEach(() => {
        wrapper.destroy()
    })

    it('it renders at all', () => {
        expect(wrapper.contains('chartist')).toBe(true)
        expect(wrapper.html()).toContain('<chartist type="Bar"')
    })

    it('has the right prop data', () => {
        expect(wrapper.props().dataModel).toContainEqual({"x": "5/15 - 6/14", "y": 13.3})
        expect(wrapper.vm.propID).toBe('chartist-bar-chart')
    })

    it('has the correct computed chart data', () => {
      expect(wrapper.vm.chartData).toHaveProperty('labels')
      expect(wrapper.vm.chartData).toHaveProperty('series')
    })

})

describe('Chartist Bar chart, snapshot', ()=>{
    let wrapper

    beforeEach(() => {
        wrapper = mount(Component, {
            propsData: {
                dataModel: data
           },
           attachToDocument: true
        })
    })

    it('matches snapshot', () => {
        let renderer = createRenderer()
        return Vue.nextTick()
            .then(function(){
                renderer.renderToString(wrapper.vm, (err, str) => {
                  if (err) throw new Error(err)
                  expect(str).toMatchSnapshot()
                })
            })

    })

})
