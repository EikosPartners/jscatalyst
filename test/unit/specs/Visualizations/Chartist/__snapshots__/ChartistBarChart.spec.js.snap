// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chartist Bar chart, snapshot matches snapshot 1`] = `
<div style="width:100%;height:100%;">
    <!---->
    <chartist type="Bar" data="[object Object]" options="[object Object]" id="chartist-bar-chart" style="height:100%;"></chartist>
</div>
`;
